### PR TITLE
Feature spec without cassettes and token

### PR DESCRIPTION
**Pull request by lavanyareddyreddy**
#### Which User story does this relate to?
-CALS-4848

#### Brief feature description
-https://osi-cwds.atlassian.net/browse/CALS-4848

#### Does this PR block/effect other stories?
- run feature spec with out being dependent on cassettes and token on any environment. 

#### Do any commands needs to be run in order to run the project with this PR?

- test without cassette and token 
npm run spec-end2end-features
(update .env with SELENIUM_BROWSER=CHROME)
- test using cassettes (current setup) 
rspec cals/spec/features (or)
SELENIUM_BROWSER=CHROME rspec cals/spec/features

add these 2 var in .env
USERNAME=
PASSWORD=

#### Code coverage percent diff as a result of this PR?
- n/a

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -48,15 +48,31 @@
...
     allow_any_instance_of(CalsBaseController).to receive(:authenticate_with_cwds).and_return(true)
     allow_any_instance_of(CalsBaseController).to receive(:get_session_token).and_return(ENV['TOKEN'])
     allow_any_instance_of(CalsBaseController).to receive(:store_token_in_redis).and_return(ENV['TOKEN'])
-    $redis.flushdb
   end
 
-
   # VCR Config
-  unless ENV['DISABLE_VCR']
+  unless ENV['TEST_END_TO_END'] || ENV['DISABLE_VCR']
```
Comment (ratneshraval) [inline spec/rails_helper.rb]: Do we need this DISABLE_VCR var? I've never used it.

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: yes if we want to use token based authentication but still want to run end to end.

**Comment by ratneshraval** [comment]
1. Can you add a script to run 'spec-end2end-features' ? 
2. CHROME and FULL_ON_CHROME can be combined as single env var SELENIUM_BROSWER, where values can be CHROME_HEADLESS / CHROME.

**Comment by lavanyareddyreddy** [comment]
combining CHROME  and FULL_ON_CHROME into SELENIUM_BROSWER sounds good.

**Comment by lavanyareddyreddy** [comment]
jenkins, retest this please.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -17,24 +17,19 @@
 
 Dir[File.dirname(__FILE__) + "/support/**/*.rb"].each {|f| require f }
 
+if ENV['SELENIUM_BROSWER'] == 'CHROME' || 'HEADLESS_CHROME'
```
Comment (Nasir2712) [inline spec/rails_helper.rb]: i know its a typo but shouldn't this be BROWSER? It's confusing

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: yup it should be BROWSER. @ratneshraval is it ok leave it or should I fix the typo. I know you have make some changes on Jenkins if I fix this typo.

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: Let me fix and commit the typo

Comment (ratneshraval) [reply spec/rails_helper.rb]: Go ahead and fix this. Also fix the scripts that uses this var.

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: fixed

Comment (Nasir2712) [reply spec/rails_helper.rb]: I have one more question. So the ENV['SELENIUM_BROSWER'] is taking in as HEADLESS_CHROME as value in package.json. Do you think ENV['SELENIUM_BROSWER']=CHROME should be in .env file? even though we are not accessing it here? Or there should be a seperate script for chrome browser altogether. Something like this
TEST_END_TO_END=true SELENIUM_BROSWER=CHROME rspec spec/features for chrome browser?

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: you can pass SELENIUM_BROSWER = HEADLESS_CHROME or SELENIUM_BROSWER = CHROME, based on how to you want to run the feature spec. whole point of test_end_to_end is to  run the feature specs without accessing cassette and token. Adding an another script TEST_END_TO_END=true SELENIUM_BROSWER=CHROME rspec spec/features is nice to have but not needed. Chrome browser is only for debugging.

Comment (lavanyareddyreddy) [reply spec/rails_helper.rb]: @Nasir2712 you can pass SELENIUM_BROSWER = HEADLESS_CHROME or SELENIUM_BROSWER = CHROME, based on how to you want to run the feature spec. whole point of test_end_to_end is to run the feature specs without accessing cassette and token. Adding an another script TEST_END_TO_END=true SELENIUM_BROSWER=CHROME rspec spec/features is nice to have but not needed. Chrome browser is only for debugging.

File: package.json [hunk 3]
```diff
@@ -60,6 +60,7 @@
     "karma-ie": "./bin/karma --browsers 'MSEdge - Win10','IE11 - Win7'",
     "all-tests": "./bin/karma;  npm run spec;",
     "spec-end2end-features": "TEST_END_TO_END=true SELENIUM_BROWSER=HEADLESS_CHROME rspec spec/features",
+    "spec-end2end-debug": "TEST_END_TO_END=true SELENIUM_BROWSER=CHROME rspec spec/features",
```
Comment (ratneshraval) [inline package.json]: @Nasir2712 Script command isn't descriptive. A suggestion `spec-end2end-features-chrome`
Maybe too long but tells you what it does

**Pull request merged**